Monika After Story/game/script-stories.rpy

```diff
@@ -51,7 +51,10 @@ init -1 python in mas_stories:
 
     #Override maps to have special conditionals for specific story types
     NEW_STORY_CONDITIONAL_OVERRIDE = {
-        TYPE_SCARY: "mas_isO31() or mas_stories.check_can_unlock_new_story(mas_stories.TYPE_SCARY)",
+        TYPE_SCARY: (
+            "(mas_isO31() and mas_stories.has_new_stories_for_type(mas_stories.TYPE_SCARY)) "
+            "or mas_stories.check_can_unlock_new_story(mas_stories.TYPE_SCARY)"

```

Comment:
I think this logic could be simplified and we could do less iterations if `check_can_unlock_new_story` could take several params to control what it checks. For example, `check_last_seen`. If `True` (default), we check the last seen of a story of this type, otherwise we skip it and only check if we have any stories.

Then this condition becomes
`mas_stories.check_can_unlock_new_story(mas_stories.TYPE_SCARY, check_last_seen=not mas_isO31())`

Naming is always a subject to change, of course. I'm not too good at this.